example-crates/example_rpc/src/main.rs

```diff
@@ -0,0 +1,308 @@
+                (None, None, None) => Auth::None,
+                (Some(path), _, _) => Auth::CookieFile(path.clone()),
+                (_, Some(user), Some(pass)) => Auth::UserPass(user.clone(), pass.clone()),
+                (_, Some(_), None) => panic!("rpc auth: missing rpc_pass"),
+                (_, None, Some(_)) => panic!("rpc auth: missing rpc_user"),
+            },
+        )?
+    };
+
+    match rpc_cmd {
+        RpcCommands::Scan {
+            fallback_height,
+            lookahead,
+            live,
+            ..
+        } => {
+            graph.lock().unwrap().index.set_lookahead_for_all(lookahead);
+
+            let (chan, recv) = sync_channel::<(EmittedUpdate, u32)>(CHANNEL_BOUND);
+            let prev_cp = chain.lock().unwrap().tip();
+
+            let join_handle = std::thread::spawn(move || -> anyhow::Result<()> {
+                let mut tip_height = Option::<u32>::None;
+
+                for item in Emitter::new(&rpc_client, fallback_height, prev_cp).into_iter() {
+                    let item = item?;
+                    let is_block = !item.is_mempool();
+                    let is_mempool = item.is_mempool();
+
+                    if tip_height.is_none() || !is_block {
+                        tip_height = Some(rpc_client.get_block_count()? as u32);
+                    }
+                    chan.send((item, tip_height.expect("must have tip height")))?;
+
+                    if sigterm_flag.load(Ordering::Acquire) {
+                        return Ok(());
+                    }
+                    if is_mempool {
+                        if !live {
+                            return Ok(());
+                        }
+                        if await_flag(&sigterm_flag, LIVE_POLL_DUR_SECS) {
+                            return Ok(());
+                        }
+                    }
+                }
+                unreachable!()
+            });
+
+            let mut start = Instant::now();
+
+            for (item, tip_height) in recv {
+                let is_mempool = item.is_mempool();
+                let tip = item.checkpoint();
+                let current_height = tip.height();
+
+                let db_changeset = {
+                    let mut indexed_additions = IndexedAdditions::default();
+                    let mut chain = chain.lock().unwrap();
+                    let mut graph = graph.lock().unwrap();
+
+                    let graph_update = {
+                        let tx_filter = bdk_bitcoind_rpc::indexer_filter(
+                            &mut graph.index,
+                            &mut indexed_additions.index_additions,
+                        );
+                        let anchor_map = bdk_bitcoind_rpc::confirmation_time_anchor;
+                        item.into_tx_graph_update(tx_filter, anchor_map)
+                    };
+                    indexed_additions.append(graph.apply_update(graph_update));
```

Comment:
This seems to be more complicated than it needs to be. Why can't I just pass off a bunch of transactions whether they be in a block or just a chunk of mempool transactions to the indexed tx graph and have it tell me the changeset after applying them. Passing off the `indexer` to the bitcoin_rpc logic seems like a hack. bitcoind_rpc shouldn't know about this type. It's a totally separate concern.

Reply:
This is addressed in 26209de743c4cd134fcd53a1db2b7c121b27230c